gtsam/hybrid/HybridBayesNet.h

```diff
@@ -181,10 +181,11 @@ class GTSAM_EXPORT HybridBayesNet : public BayesNet<HybridConditional> {
    *   auto sample = bn.sample(given, &rng);
    *
    * @param given Values of missing variables.
-   * @param rng The pseudo-random number generator.
+   * @param rng The optional pseudo-random number generator.
    * @return HybridValues
    */
-  HybridValues sample(const HybridValues &given, std::mt19937_64 *rng) const;
+  HybridValues sample(const HybridValues &given,
```

Comment:
What happens when nullptr? Seems different from yesterday's PR where we pass in a kSomething

Reply:
Passing in a static variable as a default argument is
1. Not idiomatic C++ which recommends using a `nullptr` as the default for a raw pointer.
2. Did not play well with the python wrapper due to memory allocation.

Whenever a `sample` method gets a nullptr for the `rng`, it keeps passing it along until we actually use the rng (e.g. in `Sample::sample`). At that point, I have a simple check:
```cpp
rng = (rng == nullptr) ? &kRandomNumberGenerator : rng;
```

This makes the code really clean since we don't have to worry about the rng pointer until we actually need it.

Reply:
Ok. And yesterday’s changes are now also defaulting to null ptr?

Reply:
Yes, I needed to do that to get the CI passing.